testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/EJBOverHTTPTestCase.java

```diff
@@ -0,0 +1,619 @@
+            try {
+                Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatusLine().getStatusCode());
+                log.infof("second HTTP request went to %s", response.getFirstHeader("nodename").getValue());
+            } finally {
+                HttpClientUtils.closeQuietly(response);
+            }
+        }
+    }
+
+    @InSequence(2)
+    @Test
+    public void testSLSBWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check SLSB load balance
+        testSLSBWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    @InSequence(3)
+    @Test
+    public void testSFSBWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check SFSB failover for a single SFSB
+        testSFSBWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    @InSequence(4)
+    @Test
+    public void testSFSBAndSLSBWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check SFSB and SLSB do not interfere
+        testSFSBAndSLSBWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    @InSequence(5)
+    @Test
+    public void testMultipleSFSBWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check multiple SFSB sessions are created on different nodes
+        testMultipleSFSBWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    @Ignore
+    @InSequence(6)
+    @Test
+    public void testSFSBInTransactionWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check SFSB failover for a single SFSB
+        testSFSBInTransactionWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    /*
+     * A test which checks SLSB behaviour of EJB client with EJB/HTTP in the absence of failover.
+     * The key behaviour to validate: load balancing.
+     * Proxies are obtained via JNDI/HTTP and invocations are made using EJB/HTTP.
+     */
+    public void testSLSBWithoutFailover(ExceptionSupplier<EJBDirectory, Exception> directoryProvider) throws Exception {
+
+        log.infof(MODULE_NAME+ " : testSLSBWithoutFailover: starting test");
+        try (EJBDirectory directory = directoryProvider.get()) {
+            Incrementor slsb = directory.lookupStateless(StatelessIncrementorBean.class, Incrementor.class);
+            logAffinityForBean(slsb, "testSLSBWithoutFailover");
+
+            Result<Integer> result = slsb.increment();
+            log.info("Called SLSBWithoutFailover: SLSB backend node = " + result.getNode());
+
+            int count = 1;
+            for (int i = 0; i < COUNT; ++i) {
+                result = slsb.increment();
+                log.info("Called SLSBWithoutFailover: SLSB backend node = " + result.getNode());
```

Comment:
@rachmatowicz like the `testLoadBalancer` test, this one just logs what node handles the EJB request, doesn't enforce any behavior; I guess, since we are invoking SLSB, that we expect requests to be load balanced across nodes (no sticky session)